rotkehlchen/chain/ethereum/node_inquirer.py

```diff
@@ -264,28 +261,10 @@ def assure_curve_protocol_cache_is_queried(self) -> bool:
         if should_update_protocol_cache(GeneralCacheType.CURVE_LP_TOKENS) is False:
             return False
 
-        curve_address_provider = self.contracts.contract('CURVE_ADDRESS_PROVIDER')
-        get_registry_result = curve_address_provider.call(
-            node_inquirer=self,
-            method_name='get_registry',
-        )
-        registry_address = to_checksum_address(get_registry_result)
-        registry_contract = EvmContract(
-            address=registry_address,
-            abi=self.contracts.abi('CURVE_REGISTRY'),
-            deployed_block=0,  # deployment_block is not used and the contract is dynamic
-        )
-        pools = query_curve_registry_pools(ethereum=self, registry_contract=registry_contract)  # noqa: E501
-        pools.update(query_curve_meta_pools(ethereum=self, curve_address_provider=curve_address_provider))  # noqa: E501
-        updated_pools = ensure_curve_tokens_existence(ethereum_inquirer=self, pools_mapping=pools)
-        pools_addresses = [pool_data[0] for pool_data in pools.values()]
-        gauges = query_curve_gauges(ethereum=self, registry_contract=registry_contract, known_pools=pools_addresses)  # noqa: E501
+        all_pools = query_curve_data(ethereum=self)
```

Comment:
since you return an empty list in the case of both the API & on-chain calls failing, shouldn't you check for that here?

The docstring above mentions **"returns true if the cache was modified"** but an empty list isn't modifying the cache.

Reply:
Makes sense